File: Python_rudn/task_11.py
'''
Samir Galiev, NKAbd-02-25, 10:38
Vol. 11. Task 2(upd: task 2, Vol. 8): 
Подсчитать, сколько раз встречается в заданной целочисленной матрице A(N,M) максимальное по величине число.
'''

def f(matrix):
    maxv = matrix[0][0]
    count = 0
    for row in matrix:
        for element in row:
            if element > maxv:
                maxv = element
                count = 1
            elif element == maxv:
                count += 1
    return count

n, m = map(int, input().split())
matrix = [list(map(int, input().split())) for _ in range(n)]

result = f(matrix)
print(result)

File: C++/task_11.cpp
#include <iostream>
#include <vector>
using namespace std;

int f(vector<vector<int>>& matrix) {
    int maxv = matrix[0][0];
    int count = 0;
    for (auto& row : matrix) {
        for (int element : row) {
            if (element > maxv) {
                maxv = element;
                count = 1;
            } else if (element == maxv) {
                count++;
            }
        }
    }
    return count;
}

int main() {
    int n, m;
    cin >> n >> m;
    vector<vector<int>> matrix(n, vector<int>(m));
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cin >> matrix[i][j];
        }
    }

    int result = f(matrix);
    cout << result << endl;

    return 0;
}